### PR TITLE
Added logic gates xor and xnor in Boolean Algebra

### PR DESCRIPTION
**Pull request by KuldeepBorkar**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: boolean_algebra/xnorgate.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+""" A XNOR Gate is a logic gate in boolean algebra which results to false(0)
+    if both the inputs are different and results to true(1) if both the inputs are same.
+   Following is the truth table of an XNOR Gate:
+   | Input 1 | Input 2 |  Output |
+   |      0      |     0      |      1      |
+   |      0      |     1      |      0      |
+   |      1      |     0      |      0      |
+   |      1      |     1      |      1      |
+"""
+"""Following is the code implementation of the XNOR Gate"""
```
Comment (CaedenPH) [inline boolean_algebra/xnorgate.py:10]:
```suggestion
"""
A XNOR Gate is a logic gate in boolean algebra which results to false(0)
    if both the inputs are different and results to true(1) if both the inputs are same.
   Following is the truth table of an XNOR Gate:
   | Input 1 | Input 2 |  Output |
   |      0      |     0      |      1      |
   |      0      |     1      |      0      |
   |      1      |     0      |      0      |
   |      1      |     1      |      1      |

Following is the code implementation of the XNOR Gate
"""
```

File: boolean_algebra/xnorgate.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    >>> xnor_gate(0, 0)
+    1
+    >>> xnor_gate(0, 1)
+    0
+    >>> xnor_gate(1, 0)
+    0
+    >>> xnor_gate(1, 1)
+    1
+    """
+    return int(not (input_1 != input_2))
```
Comment (CaedenPH) [inline boolean_algebra/xnorgate.py:24]:
```suggestion
    return int(input_1 == input_2)
```

File: boolean_algebra/xorgate.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+""" A XOR Gate is a logic gate in boolean algebra which results to true(1)
+    only when the two input values are different, and false(0) if the inputs are equal.
+   Following is the truth table of an XOR Gate:
+   | Input 1 | Input 2 |  Output |
+   |      0      |     0      |      0      |
+   |      0      |     1      |      1      |
+   |      1      |     0      |      1      |
+   |      1      |     1      |      0      |
+"""
+"""Following is the code implementation of the XOR Gate"""
```
Comment (CaedenPH) [inline boolean_algebra/xorgate.py:10]:
```suggestion
"""
A XOR Gate is a logic gate in boolean algebra which results to true(1)
    only when the two input values are different, and false(0) if the inputs are equal.
   Following is the truth table of an XOR Gate:
   | Input 1 | Input 2 |  Output |
   |      0      |     0      |      0      |
   |      0      |     1      |      1      |
   |      1      |     0      |      1      |
   |      1      |     1      |      0      |

Following is the code implementation of the XOR Gate
"""
```

File: boolean_algebra/xorgate.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    return int(input_1 != input_2)
+
+
+def main() -> None:
+    print("Truth Table of XOR Gate:")
+    print("| Input 1     |", " Input 2    |", " Output     |")
+    print("|      0      |", "     0      |     ", xor_gate(0, 0), "     |")
+    print("|      0      |", "     1      |     ", xor_gate(0, 1), "     |")
+    print("|      1      |", "     0      |     ", xor_gate(1, 0), "     |")
+    print("|      1      |", "     1      |     ", xor_gate(1, 1), "     |")
```
Comment (CaedenPH) [inline boolean_algebra/xorgate.py:33]: Could you use f-strings for this

File: boolean_algebra/xorgate.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    1
+    >>> xor_gate(1, 1)
+    0
+    """
+    return int(input_1 != input_2)
+
+
+def main() -> None:
+    print("Truth Table of XOR Gate:")
+    print("| Input 1     |", " Input 2    |", " Output     |")
```
Comment (CaedenPH) [inline boolean_algebra/xorgate.py:29]:
```suggestion
    print("| Input 1     |  Input 2    |  Output     |")
```

File: boolean_algebra/xnorgate.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def main() -> None:
+    print("Truth Table of XNOR Gate:")
+    print("| Input 1     |", " Input 2    |", " Output     |")
+    print("|      0      |", "     0      |     ", xnor_gate(0, 0), "     |")
+    print("|      0      |", "     1      |     ", xnor_gate(0, 1), "     |")
+    print("|      1      |", "     0      |     ", xnor_gate(1, 0), "     |")
+    print("|      1      |", "     1      |     ", xnor_gate(1, 1), "     |")
```
Comment (CaedenPH) [inline boolean_algebra/xnorgate.py:33]: Can you use f-strings for this?

File: boolean_algebra/xorgate.py (hunk 4 above)
Comment (KuldeepBorkar) [reply boolean_algebra/xorgate.py:35]: something like this?
print(f"|      0      |      0      |      {xnor_gate(0,0)}      |")
???

**Comment by KuldeepBorkar** [comment]
@CaedenPH I think I have made a mistake while updating the repo and merging conflicts and somehow that other change of Binary tree came ; ( 
I will try to fix it.

**Comment by CaedenPH** [comment]
@cclauss This has already been implemented #7596

**Comment by KuldeepBorkar** [comment]
@CaedenPH Oh! I think I actually added this earlier 15 days ago but didn't receive any reviews and now it is implemented 😿

**Pull request closed without merging**